kratos/linear_solvers/fallback_linear_solver.h

```diff
@@ -0,0 +1,897 @@
+     * which require knowledge of the spatial position of the nodes associated with a given degree of freedom (DOF).
+     * This function provides the opportunity to provide such data if needed.
+     * @param rA The sparse matrix.
+     * @param rX The solution vector.
+     * @param rB The right-hand side vector.
+     * @param rDoFSet The set of degrees of freedom.
+     * @param rModelPart The model part.
+     */
+    void ProvideAdditionalData(
+        SparseMatrixType& rA,
+        VectorType& rX,
+        VectorType& rB,
+        typename ModelPart::DofsArrayType& rDoFSet,
+        ModelPart& rModelPart
+        ) override
+    {
+        GetCurrentSolver()->ProvideAdditionalData(rA, rX, rB, rDoFSet, rModelPart);
+    }
+
+    /**
+     * @brief Add a linear solver to the collection.
+     * @details This function adds a linear solver to the collection without extending parameters.
+     * @param pSolver Pointer to the linear solver to be added.
+     */
+    void AddSolver(LinearSolverPointer pSolver)
+    {
+        // Increase the solvers vector
+        mSolvers.push_back(pSolver);
+    }
+
+    /**
+     * @brief Add a linear solver to the collection with additional parameters.
+     * @details This function adds a linear solver to the collection and extends the parameters.
+     * @param pSolver Pointer to the linear solver to be added.
+     * @param ThisParameters Parameters associated with the linear solver.
+     * @note This function extends parameters, but the interface for this is yet to be decided.
+     */
+    void AddSolver(
+        LinearSolverPointer pSolver,
+        const Parameters ThisParameters
+        )
+    {
+        // Increase the solvers vector
+        AddSolver(pSolver);
+
+        // Extend the parameters
+        FillParametersFromSolver(pSolver);
+    }
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    /**
+     * @brief Get the Solvers list.
+     * @return const std::vector<LinearSolverPointer>& Reference to the vector of solvers.
+     */
+    const std::vector<LinearSolverPointer>& GetSolvers() const
+    {
+        return mSolvers;
+    }
+
+    /**
+     * @brief Set the Solvers list.
+     * @param rSolvers A vector of LinearSolverPointer to set.
+     */
+    void SetSolvers(const std::vector<LinearSolverPointer>& rSolvers)
+    {
+        mSolvers = rSolvers;
+    }
+
+    /**
+     * @brief Check if the solver index should be reset for each try.
+     * @return true If the solver index is reset for each try.
+     * @return false Otherwise.
+     */
+    bool& GetResetSolverIndexEachTry()
+    {
+        return mResetSolverIndexEachTry;
+    }
+
+    /**
+     * @brief Set whether the solver index should be reset for each try.
+     * @param Reset Flag indicating whether to reset the solver index each try.
+     */
+    void SetResetSolverIndexEachTry(const bool Reset)
+    {
+        mResetSolverIndexEachTry = Reset;
+    }
```

Comment:
I'm not sure whether access to this member should really be exposed, but if you absolutely need to, then you should stay consistent with the rest of the functions and update the internal `Parameters`.